// @flixbus/honeycomb-react/dist/dev/esm/components/text/text.js
import {
    objectWithoutProperties as _objectWithoutProperties,
    extends as _extends
} from '../../_virtual/_rollupPluginBabelHelpers.js';
import React from 'react';
import styles from '../../node_modules/@flixbus/honeycomb/assets/scss/components/_text.js';
import HoneyBem from '../../packages/honey-bem/honey-bem.js';
import {
    Injecss
} from '../../packages/injecss/injecss.js';
var _excluded = ["Elem", "extraClasses", "children", "small"];
var bem = HoneyBem('text', styles);

var Text = function Text(_ref) {
    var Elem = _ref.Elem,
        extraClasses = _ref.extraClasses,
        children = _ref.children,
        small = _ref.small,
        rest = _objectWithoutProperties(_ref, _excluded);

    return /*#__PURE__*/ React.createElement(Elem, _extends({
        className: bem.b([{
            small: small
        }], extraClasses)
    }, rest), children);
};

Text.displayName = 'Text';
Text.defaultProps = {
    children: null,
    Elem: 'p',
    extraClasses: '',
    small: false
};
var Text$1 = Injecss(Text, styles);
export default Text$1;
export {
    Text
};
