// @flixbus/honeycomb-react/dist/dev/esm/components/heading/heading.js
import {
    objectWithoutProperties as _objectWithoutProperties,
    extends as _extends
} from '../../_virtual/_rollupPluginBabelHelpers.js';
import React from 'react';
import styles from '../../node_modules/@flixbus/honeycomb/assets/scss/components/_heading.js';
import HoneyBem from '../../packages/honey-bem/honey-bem.js';
import {
    Injecss
} from '../../packages/injecss/injecss.js';
var _excluded = ["children", "extraClasses", "flushSpace", "sectionHeader", "size"];

var Heading = function Heading(_ref) {
    var children = _ref.children,
        extraClasses = _ref.extraClasses,
        flushSpace = _ref.flushSpace,
        sectionHeader = _ref.sectionHeader,
        size = _ref.size,
        rest = _objectWithoutProperties(_ref, _excluded); // different heading type requires different HTML tags


    var Elem = "h".concat(size); // different Heading size has different block class name, e.g. `flix-h1`, `flix-h4`

    var headingBem = HoneyBem("h".concat(size), styles);
    var headingClassNames = headingBem.b({
        'section-header': sectionHeader,
        'space-flush': flushSpace
    }, extraClasses);
    return /*#__PURE__*/ React.createElement(Elem, _extends({
        className: headingClassNames
    }, rest), children);
};

Heading.displayName = 'Heading';
Heading.defaultProps = {
    extraClasses: '',
    flushSpace: false,
    sectionHeader: false
};
var Heading$1 = Injecss(Heading, styles);
export default Heading$1;
export {
    Heading
};
