// @webc/meeseeks-ui-sdk/build/esm/index.js
const extendedProductRules = {
    'TR-TR': {
        customer: {
            adult: {
                maxAmount: 10,
            },
        },
        'logged-in-customer': {
            adult: {
                maxAmount: 10,
            },
        },
    },
    'BR-BR': {
        customer: {
            adult: {
                maxAmount: 10,
            },
            br_children_0_15: {
                maxAmount: 10,
            },
        },
        'logged-in-customer': {
            adult: {
                maxAmount: 10,
            },
            br_children_0_15: {
                maxAmount: 10,
            },
        },
    },
};

const productsConfig = {
    productTypes: {
        adult: {
            type: 'adult',
            defaultAmount: 1,
            maxAmount: 42,
        },
        children: {
            type: 'child',
            defaultAmount: 0,
            maxAmount: 42,
        },
        bike_slot: {
            type: 'device',
            defaultAmount: 0,
            maxAmount: 5,
        },
        student: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        children_0_11: {
            type: 'child',
            defaultAmount: 0,
            maxAmount: 42,
        },
        children_12_15: {
            type: 'child',
            defaultAmount: 0,
            maxAmount: 42,
        },
        pensioner: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        children_0_5: {
            type: 'child',
            defaultAmount: 0,
            maxAmount: 42,
        },
        children_6_14: {
            type: 'child',
            defaultAmount: 0,
            maxAmount: 42,
        },
        pupil: {
            type: 'child',
            defaultAmount: 0,
            maxAmount: 42,
        },
        pwd: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        pl_border_guard_migration: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        pl_blind_person_guide: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        pl_war_invalid_carer: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        pl_dependent_carer: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        pl_blind_dependent: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        pl_border_guard: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        pl_customs_officer: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        pl_police_officer: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        pl_military_police: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        pl_child_disabled: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        pl_child_disabled_carer: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        pl_war_invalid_dependent: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        pl_soldier: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        pl_blind_war_victim: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        pl_student_disabled: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        pl_combatant: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        pl_blind_person: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        pl_veteran_invalid: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        pl_war_invalid: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        pl_dependent: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        pl_child_0_4: {
            type: 'child',
            defaultAmount: 0,
            maxAmount: 42,
        },
        pl_child_5_6: {
            type: 'child',
            defaultAmount: 0,
            maxAmount: 42,
        },
        ch_adult_halb_tax: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        ch_adult_general_abonnement: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        ch_children_6_15: {
            type: 'child',
            defaultAmount: 0,
            maxAmount: 42,
        },
        senior: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        children_6_17: {
            type: 'child',
            defaultAmount: 0,
            maxAmount: 42,
        },
        wheelchair: {
            type: 'device',
            defaultAmount: 0,
            maxAmount: 2,
        },
        children_0_12: {
            type: 'child',
            defaultAmount: 0,
            maxAmount: 42,
        },
        children_13_14: {
            type: 'child',
            defaultAmount: 0,
            maxAmount: 42,
        },
        children_6_11: {
            type: 'child',
            defaultAmount: 0,
            maxAmount: 42,
        },
        tr_pwd: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        children_12_17: {
            type: 'child',
            defaultAmount: 0,
            maxAmount: 42,
        },
        pl_honorary_blood_donor: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        ee_children_0_7: {
            type: 'child',
            defaultAmount: 0,
            maxAmount: 42,
        },
        ee_children_0_15_with_disability: {
            type: 'child',
            defaultAmount: 0,
            maxAmount: 42,
        },
        ee_person_with_disability: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        ee_person_with_disability_companion: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        br_pwd_free: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        br_ewli_free: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        br_ewli_half: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        br_ywli_free: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        br_ywli_half: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        br_children_0_5_free: {
            type: 'child',
            defaultAmount: 0,
            maxAmount: 42,
        },
        br_children_0_5_full: {
            type: 'child',
            defaultAmount: 0,
            maxAmount: 42,
        },
        br_children_0_15: {
            type: 'child',
            defaultAmount: 0,
            maxAmount: 42,
        },
        br_auditor: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        ua_soldier: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        ua_pitwah: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        ua_vwd: {
            type: 'adult',
            defaultAmount: 0,
            maxAmount: 42,
        },
        ua_children_lf: {
            type: 'child',
            defaultAmount: 0,
            maxAmount: 42,
        },
        ua_children_0_5: {
            type: 'child',
            defaultAmount: 0,
            maxAmount: 42,
        },
    },
    countryPassengerRules: {
        'Any-Any': ['adult', 'children', 'bike_slot'],
        'DK-DK': [
            'adult',
            'student',
            'children_0_11',
            'children_12_15',
            'pensioner',
            'bike_slot',
        ],
        'CZ-CZ': [
            'adult',
            'student',
            'children_0_5',
            'children_6_17',
            'bike_slot',
            'pwd',
            'senior',
        ],
        'CH-CH': [
            'adult',
            'children_0_5',
            'ch_children_6_15',
            'ch_adult_halb_tax',
            'ch_adult_general_abonnement',
            'bike_slot',
        ],
        'US-US': ['adult', 'children', 'bike_slot', 'wheelchair'],
        'RU-RU': ['adult', 'children_0_12', 'children_13_14', 'bike_slot'],
        'TR-TR': [
            'adult',
            'children_6_11',
            'children_12_17',
            'tr_pwd',
            'bike_slot',
        ],
        'BR-BR': ['adult', 'br_children_0_15'],
    },
    countryPassengerRulesExtended: {
        'PL-PL': [
            'pl_border_guard_migration',
            'pl_blind_person_guide',
            'pl_war_invalid_carer',
            'pl_dependent_carer',
            'pl_blind_dependent',
            'pl_border_guard',
            'pl_customs_officer',
            'pl_police_officer',
            'pl_military_police',
            'pl_child_disabled',
            'pl_child_disabled_carer',
            'pl_war_invalid_dependent',
            'pl_soldier',
            'pl_blind_war_victim',
            'pl_student_disabled',
            'pl_combatant',
            'pl_blind_person',
            'pl_veteran_invalid',
            'pl_war_invalid',
            'pl_dependent',
            'pl_honorary_blood_donor',
            'pl_child_0_4',
            'pl_child_5_6',
        ],
        'BR-BR': [
            'br_pwd_free',
            'br_ewli_free',
            'br_ewli_half',
            'br_ywli_free',
            'br_ywli_half',
            'br_children_0_5_free',
            'br_children_0_5_full',
            'br_auditor',
        ],
        'UA-UA': [
            'ua_soldier',
            'ua_pitwah',
            'ua_vwd',
            'ua_children_lf',
            'ua_children_0_5',
        ],
    },
};

const productsMapping = {
    adult: [
        'adult',
        'student',
        'pensioner',
        'pwd',
        'pl_border_guard_migration',
        'pl_blind_person_guide',
        'pl_war_invalid_carer',
        'pl_dependent_carer',
        'pl_blind_dependent',
        'pl_border_guard',
        'pl_customs_officer',
        'pl_police_officer',
        'pl_military_police',
        'pl_child_disabled_carer',
        'pl_war_invalid_dependent',
        'pl_war_invalid_dependent',
        'pl_soldier',
        'pl_blind_war_victim',
        'pl_student_disabled',
        'pl_combatant',
        'pl_blind_person',
        'pl_veteran_invalid',
        'pl_war_invalid',
        'pl_dependent',
        'pl_honorary_blood_donor',
        'ch_adult_halb_tax',
        'ch_adult_general_abonnement',
        'senior',
        'tr_pwd',
        'ee_person_with_disability',
        'ee_person_with_disability_companion',
        'br_auditor',
        'br_ewli_free',
        'br_ewli_half',
        'br_pwd_free',
        'br_ywli_free',
        'br_ywli_half',
        'ua_soldier',
        'ua_pitwah',
        'ua_vwd',
    ],
    child: [
        'children',
        'children_0_5',
        'children_0_11',
        'children_6_14',
        'children_12_15',
        'children_6_17',
        'pupil',
        'pl_child_disabled',
        'pl_child_0_4',
        'pl_child_5_6',
        'ch_children_6_15',
        'children_0_12',
        'children_13_14',
        'children_6_11',
        'children_12_17',
        'ee_children_0_7',
        'ee_children_0_15_with_disability',
        'br_children_0_5_free',
        'br_children_0_5_full',
        'ua_children_lf',
        'ua_children_0_5',
        'br_children_0_15',
    ],
    device: ['bike_slot', 'wheelchair'],
};

export {
    extendedProductRules,
    productsConfig,
    productsMapping
};
