// @flixbus/honeycomb-react/dist/dev/esm/components/header/quarks/header-brand/header-brand.js
import {
    objectWithoutProperties as _objectWithoutProperties,
    extends as _extends
} from '../../../../_virtual/_rollupPluginBabelHelpers.js';
import React from 'react';
import HoneyBem from '../../../../packages/honey-bem/honey-bem.js';
import {
    HeaderBrandPropTypes
} from './header-brand-proptypes.js';
import styles from '../../header2.js';
var _excluded = ["alt", "href", "image", "RouterLink", "square"];
var bemBrand = HoneyBem('header-brand', styles);

var HeaderBrand = function HeaderBrand(_ref) {
    var alt = _ref.alt,
        href = _ref.href,
        image = _ref.image,
        RouterLink = _ref.RouterLink,
        square = _ref.square,
        rest = _objectWithoutProperties(_ref, _excluded);

    return /*#__PURE__*/ React.createElement("div", {
        className: bemBrand.b({
            square: square
        })
    }, RouterLink ? /*#__PURE__*/ React.createElement(RouterLink, _extends({
        className: bemBrand.e('link'),
        to: href
    }, rest), /*#__PURE__*/ React.createElement("img", {
        className: bemBrand.e('img'),
        src: image,
        alt: alt
    })) : /*#__PURE__*/ React.createElement("a", _extends({
        className: bemBrand.e('link'),
        href: href
    }, rest), /*#__PURE__*/ React.createElement("img", {
        className: bemBrand.e('img'),
        src: image,
        alt: alt
    })));
};

HeaderBrand.defaultProps = {
    RouterLink: null,
    square: false
};
export {
    HeaderBrand
};
