// @flixbus/honeycomb-react/dist/dev/esm/components/header/quarks/header-navigation/header-navigation.js
import {
    objectWithoutProperties as _objectWithoutProperties,
    extends as _extends
} from '../../../../_virtual/_rollupPluginBabelHelpers.js';
import React from 'react';
import HoneyBem from '../../../../packages/honey-bem/honey-bem.js';
import {
    HeaderNavigationPropTypes
} from './header-navigation-proptypes.js';
import styles from '../../header2.js';
import NavItem from '../../../nav-side/quarks/nav-item/nav-item.js';
var _excluded = ["content", "href", "current", "key", "RouterLink"],
    _excluded2 = ["content", "href", "current", "RouterLink", "children", "key"];
var bemNav = HoneyBem('header-nav', styles);
var bemSubNav = HoneyBem('header-subnav', styles);

var buildSubNav = function buildSubNav(items) {
    return /*#__PURE__*/ React.createElement("ul", {
        className: bemSubNav.b(false)
    }, items.map(function(_ref) {
        var content = _ref.content,
            href = _ref.href,
            current = _ref.current,
            key = _ref.key,
            RouterLink = _ref.RouterLink,
            rest = _objectWithoutProperties(_ref, _excluded);

        return /*#__PURE__*/ React.createElement(NavItem, _extends({
            parentBem: bemSubNav,
            href: href,
            active: current,
            RouterLink: RouterLink,
            key: key || content
        }, rest), content);
    }));
};

var HeaderNavigation = function HeaderNavigation(_ref2) {
    var navigation = _ref2.navigation,
        ariaLabel = _ref2['aria-label'],
        extraClasses = _ref2.extraClasses;
    return /*#__PURE__*/ React.createElement("nav", {
        "aria-label": ariaLabel
    }, /*#__PURE__*/ React.createElement("ul", {
        className: bemNav.b(false, extraClasses)
    }, navigation && navigation.map(function(_ref3) {
        var content = _ref3.content,
            href = _ref3.href,
            current = _ref3.current,
            RouterLink = _ref3.RouterLink,
            children = _ref3.children,
            key = _ref3.key,
            rest = _objectWithoutProperties(_ref3, _excluded2);

        return children ? /*#__PURE__*/ React.createElement(NavItem, _extends({
            parentBem: bemNav,
            modifier: "has-subnav",
            active: current,
            subMenu: buildSubNav(children),
            openOnHover: true
        }, rest, {
            key: key || content
        }), content) : /*#__PURE__*/ React.createElement(NavItem, _extends({
            parentBem: bemNav,
            active: current,
            href: href,
            RouterLink: RouterLink
        }, rest, {
            key: key || content
        }), content);
    })));
};

HeaderNavigation.defaultProps = {
    extraClasses: '',
    navigation: [],
    'aria-label': null
};
export {
    HeaderNavigation
};
