// @flixbus/honeycomb-react/dist/dev/esm/components/skeleton/skeleton.js
import {
    defineProperty as _defineProperty
} from '../../_virtual/_rollupPluginBabelHelpers.js';
import React from 'react';
import styles from '../../node_modules/@flixbus/honeycomb/assets/scss/components/_skeleton.js';
import HoneyBem from '../../packages/honey-bem/honey-bem.js';
import {
    Injecss
} from '../../packages/injecss/injecss.js';
var bem = HoneyBem('skeleton', styles);

var Skeleton = function Skeleton(_ref) {
    var _bem$b;

    var width = _ref.width,
        height = _ref.height,
        flushBottom = _ref.flushBottom,
        extraClasses = _ref.extraClasses,
        Elem = _ref.Elem;
    return /*#__PURE__*/ React.createElement(Elem, {
        className: bem.b((_bem$b = {}, _defineProperty(_bem$b, "w-".concat(width), typeof width === 'string'), _defineProperty(_bem$b, "h-".concat(height), typeof height === 'string'), _defineProperty(_bem$b, 'flush-bottom', flushBottom === true), _bem$b), extraClasses)
    });
};

Skeleton.displayName = 'Skeleton';
Skeleton.defaultProps = {
    Elem: 'span',
    extraClasses: '',
    width: null,
    height: null,
    flushBottom: false
};
var skeleton = Injecss(Skeleton, styles);
export default skeleton;
export {
    Skeleton
};
